Realtime findQuery arrays

tstirrat opened the pull request:
Realtime queries :zap:  

Still needs tests, but seems to be working for me.

Sadly, it fetches the data twice (once to get the entire list with `.once('value', ...)`, and then again for each record due to `.on('child_added, ...)`

This would need a larger refactor and is raised in #177

/cc @halfdan

halfdan commented:
@tstirrat Will give this a try today.

halfdan commented:
Aaaaad just did. Seem to work great! Changing/Deleting/Adding entries/values is all picked up and properly changed in the store. Great work!

halfdan commented:
@tstirrat Trying to write some tests for this. Above implementation does not work correctly - find is supposed to call findQuery if the third parameter is a query (current #find only handles IDs). It might be easier to release both this and the ember-data fix PR together and release a 2.0.0. `findQuery` was just renamed to `query` (https://github.com/emberjs/data/blob/v1.13.0/packages/ember-data/lib/system/adapter.js#L157-L188) and find doesn't seem to call findQuery/query anymore.

halfdan commented:
In addition to the above: Testing actual queries w/ mockfirebase seems not possible currently: https://github.com/katowulf/mockfirebase/issues/39, any equalTo/limit/orderBy queries fail and just return all data.

tstirrat commented:
Given that ED seems to have made a whole bunch of changes, I agree that we should merge the other PR first before getting this one in.

tstirrat commented:
@halfdan The more I look at the `1.13.0` changes, the more I think we should try to land this PR before then

tstirrat commented:
Added tests, turns out the offline firebase reference handles queries fine if it knows all the data is cached locally (it is, because we set the entire hash via fixtures)

I rebased and squashed, sorry.
